### PR TITLE
Add OCP specific e2e test infra

### PR DESCRIPTION
**Pull request by creydr**
**Is this a BUG FIX or a FEATURE ?**:

> Uncomment only one, leave it on its own line:
>
> /kind bug

/kind enhancement

**What this PR does / why we need it**:
Adds the ability to run e2e tests on OCP:
* Creates some machine configs required for e2e tests on OCP
* Adds a make target to run the tests (`test-e2e-ocp`)
* Allows to run the e2e tests against clusters with OVN (sets `PRIMARY_NIC=enp1s0` in this case)

**Special notes for your reviewer**:
* This PR is based on #198 from @cybertron. 
* As adding the machine configs for persistent-networking will be added in MCO in future, this is made as a separate commit to revert
* The env vars `IMAGE_REPO` and `KUBECONFIG` must be set anyhow, e.g.: `IMAGE_REPO=creydr KUBECONFIG=~/development/dev-scripts/ocp/ostest/auth/kubeconfig make test-e2e-ocp`

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -263,4 +266,5 @@ olm-push: bundle-push index-push
 	generate-manifests \
 	tools \
 	bundle \
-	bundle-build
+	bundle-build \
+	manifests
```
Comment (cybertron) [inline Makefile:270]: Weird, this should have been handled by https://github.com/openshift/kubernetes-nmstate/pull/190

Comment (creydr) [reply Makefile]: Ah yeah. I rebased my branch and it's ok now

File: test/e2e/machineconfigs.yaml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        enabled: true
+        name: nmstate-hack.service
+
+---
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: master
+  name: 99-persistent-networking-master
```
Comment (yboaron) [inline test/e2e/machineconfigs.yaml:60]: Why do we still need the persistent networking configuration? 
didn't we handle that in MCO [1] ?

[1] https://github.com/openshift/machine-config-operator/pull/2742

**Review by cybertron — COMMENTED** [review]
Overall this looks good. Testing the latest version I have had some trouble running the tests against OVN, which doesn't seem related to any of this, and I've also had trouble with the oc wait command the first time I run this on a new cluster. The former may be a problem on my end, but we'll need to fix the latter in order to use this in CI.

File: hack/ocp-e2e-tests.sh [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+# Apply machine configs and wait until machine config pools got updated
+old_mcp_generation=$(oc get mcp master -o jsonpath={.metadata.generation})
+if oc create -f test/e2e/machineconfigs.yaml; then
+    # If MCs could be created, wait until the MCP are aware of new machine configs
+    while [ "$old_mcp_generation" -eq "$(oc get mcp master -o jsonpath={.metadata.generation})" ]; do 
+        echo "waiting for MCP update to start..."; sleep 1; 
+    done
+fi
+oc wait mcp --all --for=condition=Updated --timeout -1s
```
[the remark this answers is not in the document]
Comment (creydr) [reply hack/ocp-e2e-tests.sh]: strange you got this issue. But good you got it, before we run into it in CI. I added a loop around it as you said.

File: hack/ocp-e2e-tests.sh [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+export DEV_IMAGE_REGISTRY=quay.io
+export KUBEVIRTCI_RUNTIME=podman
+export SSH=./hack/ssh.sh
+export PRIMARY_NIC=enp2s0
+export FIRST_SECONDARY_NIC=enp3s0
+export SECOND_SECONDARY_NIC=enp4s0
+
+if oc get ns openshift-ovn-kubernetes &> /dev/null; then
+    # We are using OVNKubernetes -> use enp1s0 as primary nic
+    export PRIMARY_NIC=enp1s0
```
[the remark this answers is not in the document]
Comment (creydr) [reply hack/ocp-e2e-tests.sh:24]: I am not sure if I got you correctly. Do you think to remove this, and add it later, when we get a version that supports ovs bridges?

[the remark this answers is not in the document]
Comment (cybertron) [reply hack/ocp-e2e-tests.sh:24]: No, I'm not looking for any changes here. I was just documenting my thoughts for posterity. :-)

File: hack/ocp-e2e-tests.sh [hunk 5]
```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+
+# Make sure to set the IMAGE_REPO env variable to your quay.io username
```
Comment (jluhrsen) [inline hack/ocp-e2e-tests.sh:3]: I don't see IMAGE_REPO used below. is this comment accurate? do the 'oc' commands consume it?

**Review by yboaron — APPROVED** [review]

File: hack/ocp-e2e-tests.sh (hunk 5 above)
Comment (yboaron) [reply hack/ocp-e2e-tests.sh:3]: IMAGE_REPO used in  Makefile [1] 

[1] 
https://github.com/openshift/kubernetes-nmstate/blob/master/Makefile#L4

**Pull request merged**